File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.java
class Solution {
    public int countNegatives(int[][] grid) {
        int count=0;
        for(int i=0; i<grid.length; i++){
            int s=0, e=grid[i].length-1;
            while(s<=e){
                int m=(e+s)/2;
                if(grid[i][m]<0){
                    e=m-1;
                }else{
                    s=m+1;
                }
            }
            count+=grid[i].length-s;
        }
        return count;
    }
}

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution:
    def countNegatives(self, grid: List[List[int]]) -> int:
        c=0
        for i in range(len(grid)):
            s=int(0)
            e=int(len(grid[i])-1)
            while(s<=e):
                m=(int)(e+s)//2
                if(grid[int(i)][int(m)]<0):
                    e=m-1
                else:
                    s=m+1
            c+=(len(grid[int(i)])-int(s))
        return c